### PR TITLE
fix: Fixes nil pointer dereference in `mongodbatlas_alert_configuration`

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Fixes nil pointer dereference in `mongodbatlas_alert_configuration` when the alert has been modified through the MongoDB Atlas UI

Link to any related issue(s): https://github.com/mongodb/terraform-provider-mongodbatlas/issues/2458

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/alertconfiguration/model_alert_configuration.go
Comment (AgustinBettati) [inline internal/service/alertconfiguration/model_alert_configuration.go:206]: just to fully understand, how where you able to have the API respond with a MetricThreshold object that only had some fields defined? Which one was not being returned?

**Review by AgustinBettati — APPROVED** [review]
LGTM, changes make it safer for sure. Just want to understand more of how you reproduced the issue.

File: internal/service/alertconfiguration/model_alert_configuration.go
Comment (oarbusi) [reply internal/service/alertconfiguration/model_alert_configuration.go:206]: The issue was for `threshold_config` but I also did changes on `metric_threshold_config` just to be safe. 
The issue can be reproduced following this steps (more details [here](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/2458#issuecomment-2252309821)):
- Create an alert_notification without threshold_config 
- Through the Atlas UI, edit the alert configuration created in terraform and set a threshold_config
- terraform apply
- nil pointer dereference happens and provider crashes

**Review by maastha — APPROVED** [review]
LGTM!

File: .changelog/2463.txt
Comment (JuliaMongo) [inline .changelog/2463.txt]: Please put a period at the end of the sentence.

Comment (JuliaMongo) [reply .changelog/2463.txt]: Also, pleas consider rewriting as:

Fixes an issue where the `terraform apply` command crashes if you attempt  to edit an existing Terraform alert by adding a value to  `threshold_config`.

**Review by JuliaMongo — COMMENTED** [review]
LGTM. Please see if release note makes sense and add more detail to it if needed. Thank you!

**Comment by JuliaMongo** [comment]
LGTM after changes.

File: .changelog/2463.txt
Comment (lantoli) [inline .changelog/2463.txt]:
```suggestion
resource/mongodbatlas_alert_configuration: Fixes an issue where the `terraform apply` command crashes if you attempt to edit an existing `mongodbatlas_alert_configuration` by adding a value to `threshold_config`
```

don't finish in period, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md

same for https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2462 in case you can also change its changelog period at the end

File: internal/service/alertconfiguration/model_alert_configuration.go
Comment (lantoli) [reply internal/service/alertconfiguration/model_alert_configuration.go:206]: in these cases it would be great to create a test that fails before the fix and passes after the fix is applied

File: .changelog/2463.txt
Comment (oarbusi) [reply .changelog/2463.txt]: Okay, removing this period and other unreleased entries periods for consistency. CHANGELOG.md entries will be updated once this is merged. FYI @JuliaMongo

File: internal/service/alertconfiguration/model_alert_configuration.go
Comment (oarbusi) [reply internal/service/alertconfiguration/model_alert_configuration.go:206]: I have not been able to create a test for this because to reproduce this issue the resource has to be modified from outside TF. I don't think we currently are able to reproduce this cases in our tests. Open to suggestions though :)

**Review by lantoli — APPROVED** [review]

**Review by JuliaMongo — APPROVED** [review]
LGTM!

**Pull request merged**
